core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -138,34 +142,48 @@ func TestBodyRLPBackwardsCompatibility(t *testing.T) {
 	for _, tx := range txMatrix {
 		for _, u := range uncleMatrix {
 			for _, w := range withdrawMatrix {
-				bodies = append(bodies, &Body{tx, u, w})
+				body := &Body{
+					Transactions: tx,
+					Uncles:       u,
+					Withdrawals:  w,
+				}
+				bodies = append(bodies, body)
 			}
 		}
 	}
 
 	for _, body := range bodies {
 		t.Run("", func(t *testing.T) {
-			t.Logf("\n%s", pretty.Sprint(body))
-
 			// The original [Body] doesn't implement [rlp.Encoder] nor
 			// [rlp.Decoder] so we can use a methodless equivalent as the gold
 			// standard.
 			type withoutMethods Body
 			wantRLP, err := rlp.EncodeToBytes((*withoutMethods)(body))
-			require.NoErrorf(t, err, "rlp.EncodeToBytes([%T with methods stripped])", body)
+			require.NoErrorf(t, err, "rlp.EncodeToBytes([%T with methods stripped]) - body %s", body, pretty.Sprint(body))
 
 			t.Run("Encode", func(t *testing.T) {
 				got, err := rlp.EncodeToBytes(body)
-				require.NoErrorf(t, err, "rlp.EncodeToBytes(%#v)", body)
-				assert.Equalf(t, wantRLP, got, "rlp.EncodeToBytes(%#v)", body)
+				require.NoErrorf(t, err, "rlp.EncodeToBytes(%s)", pretty.Sprint(body))
+				assert.Equalf(t, wantRLP, got, "rlp.EncodeToBytes(%s)", pretty.Sprint(body))
 			})
 
 			t.Run("Decode", func(t *testing.T) {
 				got := new(Body)
 				err := rlp.DecodeBytes(wantRLP, got)
-				require.NoErrorf(t, err, "rlp.DecodeBytes(%v, %T)", wantRLP, got)
-
-				want := body
+				require.NoErrorf(t, err, "rlp.DecodeBytes(rlp.EncodeToBytes(%s), %T) resulted in %s",
+					pretty.Sprint(body), got, pretty.Sprint(got))
+
+				// Note we do not specify field names to enforce all fields are set.
+				gotWithoutExtra := testBodyWithoutExtra{
```

Comment:
I had to use this in order to go around cmp not handling well the unexported field extras. Feel free to give it a stab, but I have largely given up (tried many cmp options)

Note that struct is maintained in sync for exported fields with `Body` through a new unit test.

Reply:
Add `cmpopts.IgnoreUnexported(Body{})` to the options passed to `cmp.Diff()` and then you can revert everything else.